mvcc/backend/backend.go

```diff
@@ -165,31 +166,40 @@ func (b *backend) ForceCommit() {
 }
 
 func (b *backend) Snapshot() Snapshot {
+	b.batchTx.Commit()
+
+	b.mu.RLock()
+	defer b.mu.RUnlock()
+	tx, err := b.db.Begin(false)
+	if err != nil {
+		plog.Fatalf("cannot begin tx (%s)", err)
+	}
+
 	stopc, donec := make(chan struct{}), make(chan struct{})
+	dbBytes := tx.Size()
 	go func() {
 		defer close(donec)
+		// snapshotWarningTimeout is based on transferring snapshot data over a 1 gigabit/s connection
+		// assuming a min tcp throughput of 100MB/s.
+		var throughputBytes int64 = 100 * 1024 * 1014
```

Comment:
if the constants change then the getting a 0 from integer division could break this code; use floats to be safe
```go
// assume 100MB/s transfer rates
sendRateBytes := int64(100 * 1024 * 1024)
warningTimeout := time.Duration((float64(dbBytes)/float64(sendRateBytes)) * float64(time.Second))
```